pkg/test/test_data/crd-in-step/00-assert.yaml

```diff
@@ -11,8 +11,3 @@ status:
     singular: mycrd
   storedVersions:
   - v1beta1
-  conditions:
```

Comment:
We have one more condition now:
```
  - type: KubernetesAPIApprovalPolicyConformant
    reason: ApprovedAnnotation
    status: "True"
```
and with that, the test became flaky. I'm not sure if it takes a while for the transition to happen or smth. else

Reply:
Ok, so I've got to the bottom of it (which in the hindsight wasn't very deep): with three conditions, they sometimes simply come out of order and test-harness expects the same order of elements when [comparing slices in the IsSubset method](https://github.com/kudobuilder/kudo/blob/master/pkg/test/utils/subset.go#L57). A quick test confirms:
```
	assert.Nil(t, IsSubset(
		[]string{"foo", "bar"},
		[]string{"bar", "foo"},
	))
fails with: 
Expected nil, but got: &utils.SubsetError{path:[]string(nil), message:"value mismatch, expected: foo != actual: bar"}
```

However, it turns out we **can not omit** the conditions here:
- the first step creates the CRD and assumes that they're created
- the second step creates a CR of that CRD

but the second _create-step_ (**not** the assert one) occasionally fails with `the server could not find the requested resource` because the CRD hasn't gone through all transitions yet.

I'd suggest fixing `IsSubset` comparing slices but I'm not sure about all the implications. /cc @justinbarrick

Reply:
Yeah, right now it is order dependent, I need to evaluate existing tests to see if order is often important. IIRC @alenkacz and I discussed this a while back but I don't remember where we landed on it.

Reply:
Ok, so I avoided the problem for this bump by changing the CRD name/group to `prowjobs.prow.example.com` (instead of `*.k8s.io`) which doesn't require the annotation and so we only have the same two conditions as before (these two do not come out of order).  

However, the bigger question stays: I'm not sure if k8s guarantee array elements order in all places. We might want to relax the current subset comparison to be order independent or give the user the choice to use both (some sort of annotation maybe?)